### PR TITLE
shouldInitiallyConnect documentation

### PR DESCRIPTION
**Pull request by GuillaumeSalles**
Write the long overdue documentation for `shouldInitiallyConnect` documentation and fixes some typos.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -93,16 +93,16 @@ for (const other of others) {
...
 
   Usually set to false when the client is used from the server to not call the
   authentication endpoint or connect via WebSocket.
 
 In **@liveblocks/react**:
 
 - Add a new property `shouldInitiallyConnect` to `RoomProvider` that let you
-  control whether or not the room connects to Liveblock servers. Default is
+  control whether or not the room connects to Liveblocks servers. Default is
   true.
```
Comment (nvie) [inline CHANGELOG.md:106]:
```suggestion
  `true`.
```

File: docs/pages/api-reference/liveblocks-client.mdx [hunk 2]
```diff
@@ -154,6 +154,10 @@ The second argument is a configuration for the presence and storage.
   mutable by every client. Must either contain Live structures (e.g.
   `new LiveList()`, `new LiveObject({ a: 1 })`, etc.) or be serializable to
   JSON.
+- `shouldInitiallyConnect` (optional) - Whether or not the room connects to
+  Liveblocks servers. Default is `true`. Usually set to false when the client is
```
Comment (nvie) [inline docs/pages/api-reference/liveblocks-client.mdx:158]:
```suggestion
  Liveblocks servers. Default is `true`. Usually set to `false` when the client is
```

**Review by nvie — APPROVED** [review]
👍

**Comment by GuillaumeSalles** [comment]
Thanks for the review and suggestions @nvie :)

**Pull request merged**